include/astarte_bson_serializer.h

```diff
@@ -22,274 +22,264 @@
 extern "C" {
 #endif
 
-struct astarte_byte_array_t
-{
-    size_t capacity;
-    size_t size;
-    uint8_t *buf;
-};
-
-struct astarte_bson_serializer_t
-{
-    struct astarte_byte_array_t ba;
-};
+typedef struct astarte_bson_serializer *astarte_bson_serializer_handle_t;
 
 /**
  * @brief initialize given BSON serializer.
  *
  * @details This function has to be called to initialize and allocate memory for
- * astarte_bson_serializer_t.
- * @param[out] bson an astarte_bson_serializer_t that will be initialized.
+ * astarte_bson_serializer.
+ * @return The handle to the initialized astarte_bson_serializer instance.
  */
-void astarte_bson_serializer_init(struct astarte_bson_serializer_t *bson);
+astarte_bson_serializer_handle_t astarte_bson_serializer_init(void);
```

Comment:
I don't know if the function name is still valid, because it no longer initializes a `bson`, but returns a new handle of bson, so it should be changed to `new`.
```suggestion
astarte_bson_serializer_handle_t astarte_bson_serializer_new(void);
```

Reply:
Makes sense yes.